File: app/src/main/java/com/asfoundation/wallet/router/TopUpRouter.java
package com.asfoundation.wallet.router;

import android.content.Context;
import android.content.Intent;
import com.asfoundation.wallet.topup.TopUpActivity;

public class TopUpRouter {

  public void open(Context context) {
    Intent intent = TopUpActivity.newIntent(context);
    context.startActivity(intent);
  }
}

File: app/src/main/java/com/asfoundation/wallet/router/TopUpRouter.kt
package com.asfoundation.wallet.router

import android.content.Context
import android.content.Intent
import com.asfoundation.wallet.topup.TopUpActivity

class TopUpRouter {

  fun open(context: Context) {
    val intent = TopUpActivity.newIntent(context)
        .apply { flags = Intent.FLAG_ACTIVITY_SINGLE_TOP }
    context.startActivity(intent)
  }
}
